nixos/modules/services/infrastructure/terraform-cloud-agent/default.nix

```
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.terraform-cloud-agent;
+in
+{
+  options.services.terraform-cloud-agent = {
+    enable = lib.mkEnableOption "terraform-cloud-agent";
+
+    name = lib.mkOption {
+      type = lib.types.str;
+    };
+
+    log-level = lib.mkOption {
+      type = lib.types.nullOr (lib.types.enum [ "trace" "debug" "info" "warn" "error" ]);
+      default = null;
+    };
+
+    data-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    cache-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    address = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      default = null;
+    };
+
+    tokenPath = lib.mkOption {
+      type = lib.types.path;
+      default = "/etc/terraform-cloud-agent-token";
```

You can refer to the systemd configured directories via [specifiers](https://www.man7.org/linux/man-pages/man5/systemd.unit.5.html#SPECIFIERS):
```suggestion
      default = "%E/token";
```

---

Addressed in 63d4c4d6d1c5.